tests/stc_containers/leaks_check.py

```diff
@@ -0,0 +1,62 @@
+
+# pylint: disable=missing-function-docstring, missing-module-docstring
+def create_list():
+    a = [1,2,3]# pylint: disable=unused-variable
+
+def create_set():
+    a = {1,2,3}# pylint: disable=unused-variable
+
+def create_dict():
+    a = {1:1,2:2,3:3}# pylint: disable=unused-variable
+
+def list_reassign():
+    a = [1,2,3]# pylint: disable=unused-variable
+    a = [1,2,3,4]# pylint: disable=unused-variable
+    a = [i for i in range(15)]# pylint: disable=unused-variable, unnecessary-comprehension
+    a = [1,2]# pylint: disable=unused-variable
+
+def set_reassign():
+    a = {1,2,3}# pylint: disable=unused-variable
+    a = {1,2}# pylint: disable=unused-variable
+    a = {1,2,3,4}# pylint: disable=unused-variable
+    a = {1,2,3,4,5}# pylint: disable=unused-variable
+
+def dict_reassign():
+    a = {1:1,2:2,3:3}# pylint: disable=unused-variable
+    a = {1:1,2:2,3:3,4:4}# pylint: disable=unused-variable
+    a = {1:1,2:2}# pylint: disable=unused-variable
+    a = {1:1,2:2,3:3,4:4,5:5}# pylint: disable=unused-variable
+
+def conditional_list(b1: bool):
+    if (b1):
+        a = [1,2,3]# pylint: disable=unused-variable
+    a = [1,2,3,4]# pylint: disable=unused-variable
+
+def conditional_set(b1: bool):
+    if (b1):
+        a = {1,2,3}# pylint: disable=unused-variable
+    a = {1,2,3,4}# pylint: disable=unused-variable
+
+def conditional_dict(b1: bool):
+    if (b1):
+        a = {1:1,2:2,3:3}# pylint: disable=unused-variable
+    a = {1:1,2:2,3:3,4:4,5:5}# pylint: disable=unused-variable
+
+def slice_assign():
+    a = [1,2,3,4]
+    b = a[1:-1]# pylint: disable=unused-variable
```

Comment:
If there are this many unused-variables to disable it's probably better to do it for the whole file. You can add an exception to the bot

Reply:
https://github.com/pyccel/pyccel/blob/devel/ci_tools%2Fcheck_pylint_commands.py#L13

Reply:
The exception is necessary to stop the bot complaining about the file-level pylint disable command